projects/packages/jetpack-mu-wpcom/src/features/starter-page-templates/class-wp-rest-sideload-image-controller.php

```diff
@@ -0,0 +1,295 @@
+
+			/**
+			 * Fires after a single attachment is completely created or updated via the REST API.
+			 *
+			 * @param WP_Post         $attachment Inserted or updated attachment object.
+			 * @param WP_REST_Request $request    Request object.
+			 * @param bool            $creating   True when creating an attachment, false when updating.
+			 */
+			do_action( 'rest_after_insert_attachment', $attachment, $request, true );
+		}
+
+		$response = $this->prepare_item_for_response( $attachment, $request );
+		$response = rest_ensure_response( $response );
+		$response->header( 'Location', rest_url( sprintf( '%s/%s/%d', 'wp/v2', 'media', $attachment->ID ) ) );
+
+		if ( $inserted ) {
+			$response->set_status( 201 );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Creates a batch of attachments.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|\WP_REST_Response Response object on success, WP_Error object on failure.
+	 */
+	public function create_items( $request ) {
+		$data = array();
+
+		// Foreach request specified in the requests param, run the endpoint.
+		foreach ( $request['resources'] as $resource ) {
+			$request = new \WP_REST_Request( 'POST', $this->get_item_route() );
+
+			// Add specified request parameters into the request.
+			foreach ( $resource as $param_name => $param_value ) {
+				$request->set_param( $param_name, $param_value );
+			}
+
+			$response = rest_do_request( $request );
+			$data[]   = $this->prepare_for_collection( $response );
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare a response for inserting into a collection of responses.
+	 *
+	 * @param \WP_REST_Response $response Response object.
+	 * @return array|\WP_REST_Response Response data, ready for insertion into collection data.
+	 */
+	public function prepare_for_collection( $response ) {
+		if ( ! ( $response instanceof \WP_REST_Response ) ) {
+			return $response;
+		}
+
+		$data   = (array) $response->get_data();
+		$server = rest_get_server();
+
+		if ( method_exists( $server, 'get_compact_response_links' ) ) {
+			$links = call_user_func( array( $server, 'get_compact_response_links' ), $response );
+		} else {
+			$links = call_user_func( array( $server, 'get_response_links' ), $response );
+		}
+
+		if ( ! empty( $links ) ) {
+			$data['_links'] = $links;
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Prepares a single attachment output for response.
+	 *
+	 * @param \WP_Post         $post    Attachment object.
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ) {
+		$response = parent::prepare_item_for_response( $post, $request );
+		$base     = 'wp/v2/media';
+
+		foreach ( array( 'self', 'collection', 'about' ) as $link ) {
+			$response->remove_link( $link );
+
+		}
+
+		$response->add_link( 'self', rest_url( trailingslashit( $base ) . $post->ID ) );
```

Comment:
I don't understand these Static Analysis failures. I haven't found solutions so I'm going to disable all of them them with `// @phan-suppress-current-line { check name }`

Reply:
That might be a good idea for now; if you search for `PhanAccessMethodInternal` you will see that many files are already ignoring it 🥲